### PR TITLE
Fix excavation and publishing

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR

[release 2.10.0](https://github.com/palantir/resource-identifier/commit/45f90f48cce1e50a6fe703ca03c6034862265b00) [build failed to publish](https://app.circleci.com/pipelines/github/palantir/resource-identifier/1304/workflows/91b3362b-f1b8-4679-bb1e-2d4ffb3e7218/jobs/4327) and excavators are failing with the following error, causing gradle plugins and dependency bump failures.

```
> Task :jmhCompileGeneratedClasses FAILED
Compiling with JDK Java compiler API.
error: plug-in not found: ErrorProne

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jmhCompileGeneratedClasses'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':jmhCompileGeneratedClasses'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:130)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:128)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:116)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
	at org.gradle.execution.plan.DefaultPlanExecutor.process(DefaultPlanExecutor.java:116)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.executeWithServices(DefaultTaskExecutionGraph.java:138)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph.execute(DefaultTaskExecutionGraph.java:123)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:35)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:51)
	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor$ExecuteTasks.call(BuildOperationFiringBuildWorkerExecutor.java:54)
	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor$ExecuteTasks.call(BuildOperationFiringBuildWorkerExecutor.java:43)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.execution.BuildOperationFiringBuildWorkerExecutor.execute(BuildOperationFiringBuildWorkerExecutor.java:40)
	at org.gradle.internal.build.DefaultBuildLifecycleController.lambda$executeTasks$10(DefaultBuildLifecycleController.java:313)
	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:266)
	at org.gradle.internal.model.StateTransitionController.lambda$tryTransition$8(StateTransitionController.java:177)
	at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:44)
	at org.gradle.internal.model.StateTransitionController.tryTransition(StateTransitionController.java:177)
	at org.gradle.internal.build.DefaultBuildLifecycleController.executeTasks(DefaultBuildLifecycleController.java:304)
	at org.gradle.internal.build.DefaultBuildWorkGraphController$DefaultBuildWorkGraph.runWork(DefaultBuildWorkGraphController.java:220)
	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:264)
	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:128)
	at org.gradle.composite.internal.DefaultBuildController.doRun(DefaultBuildController.java:181)
	at org.gradle.composite.internal.DefaultBuildController.access$000(DefaultBuildController.java:50)
	at org.gradle.composite.internal.DefaultBuildController$BuildOpRunnable.lambda$run$0(DefaultBuildController.java:198)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.composite.internal.DefaultBuildController$BuildOpRunnable.run(DefaultBuildController.java:198)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details.
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:61)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:41)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractIsolatedCompilerWorkerExecutor$CompilerWorkAction.execute(AbstractIsolatedCompilerWorkerExecutor.java:78)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:54)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:48)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:48)
	at org.gradle.workers.internal.FlatClassLoaderWorker.run(FlatClassLoaderWorker.java:32)
	at org.gradle.workers.internal.FlatClassLoaderWorker.run(FlatClassLoaderWorker.java:22)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:102)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:71)
	at org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:146)
	at org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
	at org.gradle.process.internal.worker.request.WorkerAction.lambda$run$0(WorkerAction.java:143)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:135)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
```

## After this PR

==COMMIT_MSG==
==COMMIT_MSG==

### CODE REVIEW

File: src/test/java/com/palantir/ri/ResourceIdentifierTest.java [hunk 1]
```diff
@@ -192,70 +187,101 @@ void testSerialization() throws IOException {
...
-            prevRid = curRid;
-        }
+    @ParameterizedTest
+    @MethodSource("goodIds")
+    void testEqualsHashCode(String goodId) {
+        ResourceIdentifier curRid = ResourceIdentifier.of(goodId);
+        ResourceIdentifier curRid2 = ResourceIdentifier.of(goodId);
+        assertThat(curRid).isEqualTo(curRid2).isEqualTo(curRid);
+        assertThat(curRid2).isEqualTo(curRid).isEqualTo(curRid2);
+        assertThat(curRid).isEqualTo(curRid2);
```
Comment (felixdesouza) [inline src/test/java/com/palantir/ri/ResourceIdentifierTest.java]: redundant?

File: src/test/java/com/palantir/ri/ResourceIdentifierTest.java [hunk 2]
```diff
@@ -192,70 +187,101 @@ void testSerialization() throws IOException {
...
+
+        assertThat(curRid)
+                .hasSameHashCodeAs(curRid2)
+                .isEqualTo(curRid2)
+                .isNotSameAs(curRid2)
+                .hasSameHashCodeAs(ResourceIdentifier.of(
+                        curRid.getService(), curRid.getInstance(), curRid.getType(), curRid.getLocator()));
+
+        assertEqualsAndHashCode(curRid, curRid2);
+        assertNotEqualsAndDifferentHashCode(curRid);
```
Comment (felixdesouza) [inline src/test/java/com/palantir/ri/ResourceIdentifierTest.java]: these should just be separate tests?

File: src/test/java/com/palantir/ri/ResourceIdentifierTest.java [hunk 3]
```diff
@@ -192,70 +187,101 @@ void testSerialization() throws IOException {
...
-                "ri.service..type.name1",
-                ResourceIdentifier.of("service", "", "type", "name1").toString());
-        assertEquals(
-                "ri.service..type.name1.name2",
-                ResourceIdentifier.of("service", "", "type", "name1", "name2").toString());
-        assertEquals(
-                "ri.service..type.name1.name2.name3",
-                ResourceIdentifier.of("service", "", "type", "name1", "name2", "name3")
-                        .toString());
+        assertThat(ResourceIdentifier.of("service", "", "type", "name1").toString())
```
Comment (felixdesouza) [inline src/test/java/com/palantir/ri/ResourceIdentifierTest.java]: you could just do `.hasToString`, you use the inverse in `assertNotEqualsAndDifferentHashCode` below

File: src/test/java/com/palantir/ri/ResourceIdentifierTest.java [hunk 4]
```diff
@@ -192,70 +187,101 @@ void testSerialization() throws IOException {
...
+    @ParameterizedTest
+    @MethodSource("goodIds")
+    void testEqualsHashCode(String goodId) {
+        ResourceIdentifier curRid = ResourceIdentifier.of(goodId);
+        ResourceIdentifier curRid2 = ResourceIdentifier.of(goodId);
+        assertThat(curRid).isEqualTo(curRid2).isEqualTo(curRid);
+        assertThat(curRid2).isEqualTo(curRid).isEqualTo(curRid2);
+        assertThat(curRid).isEqualTo(curRid2);
+        assertThat(curRid.toString()).isEqualTo(curRid2.toString());
+        assertThat(curRid.hashCode()).isEqualTo(curRid2.hashCode());
```
Comment (felixdesouza) [inline src/test/java/com/palantir/ri/ResourceIdentifierTest.java]: redundant from line 241

File: src/test/java/com/palantir/ri/ResourceIdentifierTest.java [hunk 5]
```diff
@@ -192,70 +187,101 @@ void testSerialization() throws IOException {
...
+        assertThat(curRid).isNotEqualTo(NotEqualsObj.INSTANCE);
+
+        assertThat(curRid)
+                .hasSameHashCodeAs(curRid2)
+                .isEqualTo(curRid2)
```
Comment (felixdesouza) [inline src/test/java/com/palantir/ri/ResourceIdentifierTest.java]: redundant from lines 233 and 234

**Review by felixdesouza — COMMENTED** [review]
assertj makes it much nicer! small comments however

**Comment by schlosna** [comment]
> assertj makes it much nicer! small comments however

thanks, cleaned up the tests a bit more and added coverage for a few more edge cases

**Review by felixdesouza — APPROVED** [review]
